Fix bug with value transfer

macfarla opened the pull request:
Signed-off-by: Sally MacFarlane <sally.macfarlane@consensys.net>

Clarify isGoQuorumPrivateTransaction method. v value is set directly for GoQuorum private transactions, and they don't have a chainId. All other tx derive their v value from the chainId.

fixes #2606 

## Changelog

- [x] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

garyschulte reviewed (approved):
LGTM!

lucassaldanha reviewed (changes requested):
We also need to fix the TransactionPool.

https://github.com/hyperledger/besu/blob/main/ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java#L294-L297

It would be better to consolidate the logic into a single place.

macfarla commented:
> We also need to fix the TransactionPool.
> 
> https://github.com/hyperledger/besu/blob/main/ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java#L294-L297
> 
> It would be better to consolidate the logic into a single place.

Updated. And added a TransactionPool test. Thanks @lucassaldanha
